WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizer.swift

```diff
@@ -335,17 +354,22 @@ private extension RemoteOrderSynchronizer {
     /// Returns a publisher that creates an order remotely, configured for the given operation type.
     /// The later emitted order is delivered with the latest selected status.
     ///
-    func createOrderRemotely(_ order: Order, type: OperationType) -> AnyPublisher<Order, Error> {
+    func createOrderRemotely(_ order: Order, type: OperationType, includesGiftCard: Bool) -> AnyPublisher<Order, Error> {
         Future<Order, Error> { [weak self] promise in
             guard let self = self else { return }
 
+            let giftCard = includesGiftCard ? self.giftCardToApply: nil
+
             let apiOrderStatus = self.orderStatus(for: type)
             let draftOrder = order.copy(status: apiOrderStatus).sanitizingLocalItems()
-            let action = OrderAction.createOrder(siteID: self.siteID, order: draftOrder) { [weak self] result in
+            let action = OrderAction.createOrder(siteID: self.siteID, order: draftOrder, giftCard: giftCard) { [weak self] result in
                 guard let self = self else { return }
 
                 switch result {
                 case .success(let remoteOrder):
+                    if giftCard != nil {
+                        self.giftCardToApply = nil
+                    }
```

Comment:
I haven't touched the OrderSynchronizer in a while (Since it's v1) but I don't quite understand why this has to be nilled when other fields arent 😅

Reply:
Other fields are part of the `Order` object, and are updated in the Yosemite layer when creating/updating the order remotely. The gift card code is a separate state variable in the order form, and needs to be cleared out separately when the gift card is applied successfully.

Lemme know if I might be missing anything!